R/execute.R

```diff
@@ -83,293 +123,584 @@ rmf_run_modflow.modflow <- function(modflow,
+    rmf_install_bin_folder <- paste0(getOption("RMODFLOW.path"), "/", code,
+                                     "/bin/")
     if (!file.exists(executable)) {
       if (file.exists(paste0(rmf_install_bin_folder, executable))) {
         folder <- rmf_install_bin_folder
       } else if (Sys.which(executable) == "") {
-        stop("Path to ", name, " executable not found.")
+        rui::stop("Path to {code} executable not found.")
       }
     }
     return(paste0(folder, executable))
-  } else if (name == "MODFLOW-LGR") {
+  } 
+  if (grepl("lgr", code, ignore.case = TRUE)) {
+    code <- "MODFLOW-LGR"
     executable <- "mflgr.exe"
     folder <- ""
-    rmf_install_bin_folder <- paste0("C:/WRDAPP/", name, "/bin/")
+    rmf_install_bin_folder <- paste0(getOption("RMODFLOW.path"), "/", code,
+                                     "/bin/")
     if (!file.exists(executable)) {
       if (file.exists(paste0(rmf_install_bin_folder, executable))) {
         folder <- rmf_install_bin_folder
       } else if (Sys.which(executable) == "") {
-        stop("Path to ", name, " executable not found.")
+        rui::stop("Path to {code} executable not found.")
       }
     }
     return(paste0(folder, executable))
-  } else if (name == "MODFLOW-CFP") {
+  } 
+  if (grepl("cfp", code, ignore.case = TRUE)) {
+    code <- "MODFLOW-CFP"
     executable <- "mf2005cfp.exe"
     folder <- ""
-    rmf_install_bin_folder <- paste0("C:/WRDAPP/", name, "/")
+    rmf_install_bin_folder <- paste0(getOption("RMODFLOW.path"), "/", code,
+                                     "/bin/")
     if (!file.exists(executable)) {
       if (file.exists(paste0(rmf_install_bin_folder, executable))) {
         folder <- rmf_install_bin_folder
       } else if (Sys.which(executable) == "") {
-        stop("Path to ", name, " executable not found.")
+        rui::stop("Path to {code} executable not found.")
       }
     }
     return(paste0(folder, executable))
+  }
+  rui::alert("Finding paths to the executables of codes other than ",
+           "MODFLOW-2005, MODFLOW-OWHM, MODFLOW-NWT, MODFLOW-LGR or ",
+           "MODFLOW-CFP is currently not supported.")
+  rui::stop("Issue with code path.")
+}
+
+#' Look for a file path in a NAM file
+#'
+#' @param dir Character. Path to directory containing the NAM file.
+#' @param nam Character. File name of the NAM file.
+#' @param type Character. File type (ftype) of the entry to look for.
+#' @param unit Integer. Unit number of the entry to look for.
+#' @return
+rmfi_look_for_path <- function(dir, nam, type = NULL, unit = NULL) {
+  if (!is.null(type)) {
+    return(file.path(dir, nam$fname[which(nam$ftype == stringr::str_to_upper(type))]))
+  }
+  if (!is.null(unit)) {
+    return(file.path(dir, nam$fname[which(nam$nunit == unit)]))
+  }
+  rui::alert("Either {.arg type} or {.arg unit} should be provided.")
+  rui::stop("Issue with arguments.")
+}
+
+rmfi_line_callback <- function(line, process) {
```

Comment:
What does this do exactly?

Reply:
This function processes the stdout output from MODFLOW, using the {rui} UI functions, so that MODFLOW seems to be more nicely integrated in the R console/RMODFLOW package. It is used in the `processx::run()` call that launches the MODFLOW executables.